src/Components/controlOption.js

```diff
@@ -3,23 +3,39 @@ import { Checkbox } from "semantic-ui-react";
 
 const controlOption = (props, controlSettings, toggleLayer, i) => {
   return (
-    <div className={
-      "controller-option " +
-      ( controlSettings.dep ? 'option-secondary-checkbox ' : '' ) +
-      ( controlSettings.dep && !props[ controlSettings.dep ] ? 'disabled' : '' )
-    } key={ i }>
+    <div
+      className={
+        "controller-option " +
+        (controlSettings.indent ? "option-indented-checkbox " : "") +
+        (shouldDisable(props, controlSettings.dep) ? "disabled" : "")
+      }
+      key={i}
+    >
       <Checkbox
-        label={ controlSettings.label }
-        ref={ controlSettings.ref }
+        label={controlSettings.label}
+        ref={controlSettings.ref}
         type="checkbox"
-        defaultChecked={ props[ controlSettings.key ] }
-        onChange={ (e, { checked }) => toggleLayer({
-          ...controlSettings,
-          checked
-        }) }
+        checked={setChecked(props, controlSettings)}
+        defaultChecked={props[controlSettings.key]}
+        onChange={(e, { checked }) =>
+          toggleLayer({
+            ...controlSettings,
+            checked
+          })
+        }
       />
     </div>
-  )
+  );
+};
+
+function shouldDisable(props, dependentBox) {
+  return dependentBox && !props[dependentBox];
+}
+
+function setChecked(props, controlSettings) {
+  return controlSettings.isMaster
+    ? props.showPublicTransit

```

Comment:
This will break if we have another master checkbox control. Maybe not an issue right now.

Reply:
We don't need this anymore. Removed the method.